learningequality/studio
# Add image resizing functionality for perseus file exports

## Summary
### Description of the change(s) you made
This PR introduces image resizing functionality to resize assessment images containing resizing data and making sure that the final perseus zip contains resized images.


### Manual verification steps performed
1. Creating assessments containing varied image data, like original, resized and duplicate images.
2. Publishing the channel and testing the changes on local Kolibri instance by importing it from the locally running studio server.

___
## Reviewer guidance
### How can a reviewer test these changes?
1. Follow the verification steps listed under the previous heading.
2. Create a dummy exercise and try adding images and resizing them in the exercise questions.
3. Publish the channel and import it in Kolibri(Use local studio for imports by changing the environment variable `CENTRAL_CONTENT_BASE_URL` to point to the URL of your local studio instance).
4. Checkout the imported assessment to see if images have been resized accordingly or not.

## References
Closes #4416 

## Comments
The approach used here slightly deviates from what was told in the issue description. While working on this issue, I found out that the control flow of the channel publish method never entered the image loop in `create_perseus_zip` function. So, the image resizing function couldn't be applied there. Instead, the image resizing now happens in the `process_image_strings` function as this is where the image data is modified originally before being written to the assessment file and it contains the image resizing data too.

----

### Contributor's Checklist


PR process:


Studio-specifc:



Testing:

___

### Reviewer's Checklist
#### This section is for reviewers to fill out.
